lhotse/recipes/ami/diarization.py

```diff
@@ -0,0 +1,191 @@
+"""
+The data prepare recipe for the AMI Meeting Corpus diarization task.
+
+In this recipe, the IHM-Mix and SDM settings are supported.
+
+We use the Full Partition (used in Kaldi s5c recipe) which has more dev and eval sessions than the ASR partition, 
+and references are obtained from the ami_manual_annotations_v1.6.2. We get both the split and the references
+from https://github.com/BUTSpeechFIT/AMI-diarization-setup
+"""
+
+from collections import defaultdict
+
+import logging
+import os
+import re
+import soundfile
+import urllib.request
+from pathlib import Path
+
+from typing import Dict, List, NamedTuple, Optional, Union
+
+from lhotse.qa import validate_recordings_and_supervisions
+from lhotse.audio import AudioSource, Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike, Seconds
+
+from lhotse.recipes.ami.common import download_audio, prepare_audio_other, remove_supervision_exceeding_audio_duration
+
+# TODO: support the "ihm" and "mdm" microphone settings
```

Comment:
Do you intend to add these too? (it's ok to say no)

Reply:
Yes I'm adding them now in the next commit.